genrl/deep/agents/a2c/a2c.py

```diff
@@ -116,6 +123,27 @@ def create_model(self) -> None:
 
         self.rollout = RolloutBuffer(self.rollout_size, self.env)
 
+    def create_custom_model(self) -> None:
+        """
+        Creates custom actor critic model and initialises optimizers
+        """
+        if self.noise is not None:
+            self.noise = self.noise(
+                np.zeros_like(action_dim), self.noise_std * np.ones_like(action_dim)
+            )
+        try:
+            self.ac = self.network_type(**kwargs).to(self.device)
```

Comment:
Seems to me this is the only difference between `create_model` and `create_custom_model`. Can't you use an if statement here instead?
```
if isinstance(self.network_type, str):
    self.ac = get_model(...)
else:
    self.ac = self.network_type(...)
```

Reply:
Used this, increased the cognitive comlplexity of sac.py to 8, but all the others are fine. I'm use `create_custom_model()` for sac.py, and go with this for others. Would cause some non-uniformity, thats okay right?

Reply:
No it should be fine, SAC needs to be refactored so it can be handled later on.